R/derive_param_diam_to_sev.R

```diff
@@ -0,0 +1,224 @@
+#' 5.User can pass the events which has diameter values to derive their severity
+#'  records by passing the appropriate `filter_diam`, `filter_faobj`and severity
+#'  grade limits.
+#'
+#' @return
+#' The Input data with the new severity records for Redness and swelling which
+#' is specified in `filter_faobj` and AVAL, AVALC will be derived and fatestcd,
+#'  fatest will be changed as per the values
+#'
+#' @examples
+#' library(dplyr)
+#' library(admiral)
+#' library(tibble)
+#'
+#' input <- tribble(
+#'   ~USUBJID, ~FAOBJ, ~AVAL, ~AVALC, ~ATPTREF, ~FATEST, ~FATESTCD,
+#'   "XYZ1001", "REDNESS", 7.5, "7.5", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "REDNESS", 3.5, "3.5", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "REDNESS", 2, "2", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "REDNESS", 1.8, "1.8", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "REDNESS", 1.4, "1.4", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 11.1, "11.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 7.4, "7.4", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 6, "6", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 2.1, "2.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 1.1, "1.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 5.5, "5.5", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 2.5, "2.5", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 2, "2", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 1.8, "1.8", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 1.4, "1.4", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 10.1, "10.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 7.1, "7.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 5, "5", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 1.8, "1.8", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 1.4, "1.4", "VACCINATION 2", "Diameter", "DIAMETER"
+#' )
+#'
+#' derive_param_diam_to_sev(
+#'   dataset = input,
+#'   filter_faobj = c("REDENSS", "SWELLING"),
+#'   filter_diam = "DIAMETER",
+#'   testcd_sev = "SEV",
+#'   test_sev = "Severity"
+#' )
+#'
+#' @export
+#' @keywords der_adxx
+#' @family der_adxx
+#'
+derive_param_diam_to_sev <- function(dataset = NULL,
+                                     filter_diam = "DIAMETER",
+                                     filter_faobj = c("REDNESS", "SWELLING"),
+                                     testcd_sev = "SEV",
+                                     test_sev = "Severity/Intensity",
+                                     none = c(0, 2),
+                                     mild = c(2, 5),
+                                     mod = c(5, 10),
+                                     sev = 10) {
```

Comment:
Could a user put any numbers in here?  Do these need to get check more robustly?

Reply:
Yes! User can put any number. Basically, for assigning the severity grade in AVALC, we usually pass the values as unequal class intervals.
for 'sev' argument user should pass a single value. (aval > sev) the values in AVAL which is greater than that 'sev' value is considered as "SEVERE".
I have explained these things in Roxygen comments. 
And.. Yes! it will work for any robust values if those values are existed in the AVAL.